src/platform/forms-system/src/js/review/ReviewFieldTemplate.jsx

```diff
@@ -14,16 +14,16 @@ export default function ReviewFieldTemplate(props) {
     : null;
   const pageIndex = formContext?.pagePerItemIndex;
 
-  if (schema.type === 'object' || schema.type === 'array') {
-    return children;
-  }
-
   // `hideEmptyValueInReview` option is ignored if a 'ui:reviewField' is defined
   // The custom reviewField should handle empty values
   if (uiSchema?.['ui:reviewField']) {
     return uiSchema['ui:reviewField'](props);
   }
```

Comment:
Need to 100% verify that this logic reordering does not cause any degradation in production forms. Theoretically, this would only impact forms using `ui:reviewField` in the wrong way. Prior to this change, using `ui:reviewField` on an object or array would not change anything about the rendering on the review page. Now, `ui:reviewField` will take the priority. So, the only locations this would impact are places where teams were accidentally using `ui:reviewField` and not experiencing any changes because of the prior logic.

Reply:
Verified there are no objects or arrays using `ui:reviewField`, so this change should not cause any regressions.